### DISCUSSION TITLE
Method Not Allowed

### DISCUSSION THREAD

**Discussion by Link0Darck** [cockpit-project/cockpit#18767] [General]
### Explain what happens

Hello,
I am using Reverse Proxy to access my server from my domain name.
I followed this tutorial except the fact of putting /cp/ because it is useless for me.

Tutorial link: [Tutorial](https://evermeet.cx/wiki/Reverse_Proxy_setup_for_Cockpit_with_Apache_(in_a_sub_directory))

What I don't understand is that sometimes it works and sometimes it gives me an error: **Method Not Allowed**

Example of what I put `Origins = https://example.com http://127.0.0.1:9090 http://192.168.60.193:9090`
For information example.com I put by my domain name but I still get the error

I want to access with my domain name and locally too and I use apache2 reverse proxy

Thanks in advance for the help

### Version of Cockpit

Version 239.

### Where is the problem in Cockpit?

All

### Server operating system

Debian

### Server operating system version

Debian GNU/Linux 11 (bullseye) aarch64

### What browsers are you using?

OperaGX

### System log

```shell
journalctl --since -10m :
https://pastebin.com/raw/K7aR5KYV

journalctl -u cockpit -u <unit> :
https://pastebin.com/raw/DxYQpxCX
```

**Comment by martinpitt** [comment]
cockpit-ws[285735]: received unsupported HTTP method

My guess is that your reverse proxy tries to speak HTTP/2 or even /3 to cockpit-ws -- that doesn't work, our web server only supports 1.1. Although "method" sounds more like GET vs. POST etc -- cockpit's web server only speaks GET and HEAD, not POST/PUT/DELETE.

As you don't need https between the proxy and cockpit-ws, you can try to `systemctl stop cockpit.socket cockpit` and run cockpit-ws manually in debugging mode:

     sudo -u cockpit-wsinstance env G_MESSAGES_DEBUG=all /usr/lib/cockpit/cockpit-ws --no-tls -a 127.0.0.1

Then you should get some more output what happens.

**Reply by Link0Darck** [reply]
> ```
> cockpit-ws[285735]: received unsupported HTTP method
> ```
> 
> My guess is that your reverse proxy tries to speak HTTP/2 or even /3 to cockpit-ws -- that doesn't work, our web server only supports 1.1. Although "method" sounds more like GET vs. POST etc -- cockpit's web server only speaks GET and HEAD, not POST/PUT/DELETE.
> 
> As you don't need https between the proxy and cockpit-ws, you can try to `systemctl stop cockpit.socket cockpit` and run cockpit-ws manually in debugging mode:
> 
> ```
>  sudo -u cockpit-wsinstance env G_MESSAGES_DEBUG=all /usr/lib/cockpit/cockpit-ws --no-tls -a 127.0.0.1
> ```
> 
> Then you should get some more output what happens.

Hello,
I did the command you told me but it doesn't work and I even disabled HTTP/2 and 3 but it doesn't work. I even looked in cloudflare to disable the parameters
